sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreClient.java

```diff
@@ -172,10 +171,10 @@ private void updateResponseHeader(RxDocumentServiceRequest request, Map<String,
             return;
         }
 
-        String partitionKeyRangeId = headers.get(WFConstants.BackendHeaders.PARTITION_KEY_RANGE_ID);
+        String partitionKeyRangeId = headers.getValue(WFConstants.BackendHeaders.PARTITION_KEY_RANGE_ID);
```

Comment:
perf impact?

any invocation to `HttpHeaders.getValue(name)` will invoke name.toLower
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java#L81

Reply:
Best way forward is to do perf testing and see the impact. 
And we can always follow-up on the optimization if the impact is with-in the bounds.

Reply:
Check the perf numbers below.